cleverhans/attacks.py

```diff
@@ -1182,12 +1182,13 @@ def attack_single_step(self, x, eta, y):
         grad, = tf.gradients(loss, adv_x)
         scaled_signed_grad = self.eps_iter * tf.sign(grad)
         adv_x = adv_x + scaled_signed_grad
-        adv_x = tf.clip_by_value(adv_x, self.clip_min, self.clip_max)
+        if self.clip_min is not None and self.clip_max is not None:
+            adv_x = tf.clip_by_value(adv_x, self.clip_min, self.clip_max)
         eta = adv_x - x
         eta = clip_eta(eta, self.ord, self.eps)
         return x, eta
 
-    def attack(self, x, **kwargs):
+    def attack(self, x, y, **kwargs):
```

Comment:
It would be best to pass y as one of the kwargs, to keep the unified interface across different attacks.

Reply:
I think you are comparing to the interface for `generate` function but this line defines the `attack` function which is special to the MadryEtAl and similar to function `fgm` here: https://github.com/tensorflow/cleverhans/blob/cca980212c07b952d0849e3df4d59b6185e3ce91/cleverhans/attacks_tf.py#L23

The kwargs was unnecessary and I removed it in 0bc4a67.